File: match-api/src/main/java/com/piter/match/api/domain/Match.java
package com.piter.match.api.domain;

import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Value;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.mongodb.core.mapping.Document;

@Value
@Builder
@Document
public class Match {

  @Transient
  public static final String SEQUENCE_NAME = "match_sequence";

  @Id
  Long id;
  String homeTeam;
  String awayTeam;
  LocalDateTime startTime;
  MatchResult result;
  MatchRound round;
}

File: bet-frontend/src/app/model/match.ts
export interface Match {
  id: number;
  homeTeam: string;
  awayTeam: string;
  startTime: Date;
  result: MatchResult | null;
  round: MatchRound;
}

export class MatchResult {
  homeTeamGoals: number;
  awayTeamGoals: number;

  constructor(homeTeamGoals: number, awayTeamGoals: number) {
    this.homeTeamGoals = homeTeamGoals;
    this.awayTeamGoals = awayTeamGoals;
  }
}

export interface MatchRound {
  roundName: string;
  startTime: Date;
}
